### ISSUE TITLE
Haproxy metric error

### ISSUE THREAD

**Issue by bgfor** [netdata/netdata#6841]
I have a problem with collecting statistics from haproxy.
Until yesterday's update, netdata worked fine.
I see an error in the log when starting:
```
2019-09-13 10:33:23: python.d ERROR: haproxy[haproxy] : URL is not defined or type is not <str>
2019-09-13 10:33:23: python.d INFO: plugin[main] : haproxy[haproxy] : check failed
```

Netdata Config file:
```
#via_url:
 user : 'admin'
 pass : 'password'
 url : 'http://127.0.0.1:7000/haproxy_stats;csv;norefresh'
```

HAProxy config:
```
listen  stats-for-netdata
        bind 127.0.0.1:7000
        mode    http
        no      log
        maxconn 10
        stats   enable
        stats   auth admin:password
        stats   uri  /haproxy_stats

```

`curl -u admin:password 'http://127.0.0.1:7000/haproxy_stats;csv;norefresh'` works fine and returns values.

**Comment by cakrit** [comment]
This could only be due to #6661 . Can you please [debug the module](https://docs.netdata.cloud/collectors/python.d.plugin/#how-to-debug-a-python-module) and tell us what you see?

**Comment by bgfor** [comment]
```
netdata@eblade1:/opt/netdata/usr/libexec/netdata/plugins.d$ ./python.d.plugin haproxy debug trace
2019-09-13 12:53:33: python.d INFO: plugin[main] : using python v2
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : looking for 'python.d.conf' in ['/opt/netdata/etc/netdata', '/opt/netdata/usr/lib/netdata/conf.d']
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : loading '/opt/netdata/etc/netdata/python.d.conf'
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : '/opt/netdata/etc/netdata/python.d.conf' is loaded
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : looking for 'pythond-jobs-statuses.json' in /opt/netdata/var/lib/netdata
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : loading '/opt/netdata/var/lib/netdata/pythond-jobs-statuses.json'
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : '/opt/netdata/var/lib/netdata/pythond-jobs-statuses.json' is loaded
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : [haproxy] looking for 'haproxy.conf' in ['/opt/netdata/etc/netdata/python.d', '/opt/netdata/usr/lib/netdata/conf.d/python.d']
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : [haproxy] loading '/opt/netdata/etc/netdata/python.d/haproxy.conf'
2019-09-13 12:53:33: python.d DEBUG: plugin[main] : [haproxy] '/opt/netdata/etc/netdata/python.d/haproxy.conf' is loaded
2019-09-13 12:53:33: python.d INFO: plugin[main] : [haproxy] built 1 job(s) configs
2019-09-13 12:53:33: python.d DEBUG: haproxy[haproxy] : urllib3 version: 1.21.1
2019-09-13 12:53:33: python.d ERROR: haproxy[haproxy] : URL is not defined or type is not <str>
2019-09-13 12:53:33: python.d ERROR: haproxy[haproxy] : Traceback (most recent call last):
  File "./python.d.plugin", line 25, in <module>
    from queue import Queue
ImportError: No module named queue

2019-09-13 12:53:33: python.d INFO: plugin[main] : haproxy[haproxy] : check failed
2019-09-13 12:53:33: python.d INFO: plugin[main] : no jobs to serve
2019-09-13 12:53:33: python.d INFO: plugin[main] : exiting from main...

```

**Comment by cakrit** [comment]
I think I know what's happening. Can you please `cat /opt/netdata/var/lib/netdata/pythond-jobs-statuses.json` and show the results here? I have the hunch that we need to escape some characters to make it a proper JSON file.

**Comment by bgfor** [comment]
```
{
  "nginx": {
    "localhost": "active"
  },
  "ntpd": {
    "localhost": "active"
  },
  "postfix": {
    "local": "active"
  },
  "web_log": {
    "nginx_log": "active"
  }
}
```

**Comment by cakrit** [comment]
Then it's not what I suspected. Can you just verify that it is `/opt/netdata/etc/netdata/python.d/haproxy.conf` that has the following?

``
 user : 'admin'
 pass : 'password'
 url : 'http://127.0.0.1:7000/haproxy_stats;csv;norefresh'
```

**Comment by bgfor** [comment]
Yes it does.
I didn't change anything in the configuration, only updated the netdata software.

```
# ----------------------------------------------------------------------
# AUTO-DETECTION JOBS
# only one of them will run (they have the same name)

#via_url:
 user : 'admin'
 pass : 'password'
 url : 'http://127.0.0.1:7000/haproxy_stats;csv;norefresh'

#via_socket:
# socket: '/var/run/haproxy/admin.sock'
```

`curl -u admin:password 'http://127.0.0.1:7000/haproxy_stats;csv;norefresh'` works fine and returns values.

**Comment by cakrit** [comment]
Never mind, @vlvkobal has reproduced it and is looking into it.

**Comment by ilyam8** [comment]
There was a bug in parsing configuration files w/o explicitly defined jobs.

Multi job config:

```yaml
job_name_1:
 arg : 'value'

job_name_2:
 arg : 'value'

job_name_3:
 arg : 'value'
```

There is a special case - no job config:

```yaml
 arg1 : 'value'
 arg2 : 'value'
 arg2 : 'value'
```

The bug was in parsing special case. Fixes in #6856
